CMakeLists.txt

```diff
@@ -486,6 +504,20 @@ function(hermes_link_icu target_name)
   endif()
 endfunction()
 
+# Declare a function that strips debugging symbols from any target built for
+# Apple platforms, which we do build initially so they can be extracted into a
+# dSYM bundle.
+function(hermes_strip_debug_symbols target_name)
```

Comment:
I'm slightly confused by the fact that `hermes_strip_debug_symbols` only strips symbols on Apple. Should the name indicate this, or should there be a separate `STRIP_OUTPUTS` option to control this (e.g. running `xcrun strip` on Apple, fail on Windows, and `strip` on other OS)?

Reply:
I'm in the process of making changes to save unstripped libraries on Android, and this scared me for a moment. Since this seems to be specific to making apple dSYMs work, I think my preference here would be to rename this to make it clear it's about Apple, and we can address other platforms separately, instead of including it in this PR.